src/renderer/overview.tsx

```diff
@@ -1,29 +1,38 @@
 import React from 'react';
 import { Banner } from '@getflywheel/local-components';
 import LastOptimizeStatus from './lastOptimizeStatus';
+import { store, selectors } from './store';
 import { SiteImageData } from '../types';
 
 interface IProps {
 	setOverviewSelected: (x: boolean) => void,
 	handleScanForImages: () => void,
-	scanImageState: SiteImageData,
-	fetchImageStateData: () => void,
+	siteImageData: SiteImageData,
+	siteID: string,
 }
 
 export const Overview = (props: IProps) => {
-	const { setOverviewSelected, handleScanForImages, scanImageState } = props;
 	const {
-		imageCount,
-		totalCompressedCount,
+		setOverviewSelected,
+		handleScanForImages,
+		siteImageData,
+	} = props;
+
+	const {
 		lastScan,
-		scanInProgress
-	} = scanImageState;
+		scanInProgress,
+	} = siteImageData;
 
 	const onClickViewImages = () => {
 		setOverviewSelected(false);
-		props.fetchImageStateData();
 	}
 
+	const imageCount = selectors.siteImageCount(store.getState(), props);
+	const originalTotalSize = selectors.totalImagesSizeBeforeCompression(store.getState(), props);
+	const compressedImagesOriginalSize = selectors.originalSizeOfCompressedImages(store.getState(), props);
+	const compressedTotalSize = selectors.sizeOfCompressedImages(store.getState(), props);
+	const totalCompressedCount = selectors.compressedSiteImages(store.getState(), props).length;
+
```

Comment:
I just love to see all the selectors you've setup!! In terms of getting this data, I wonder if wrapping all of this up in one more selector that returns all of these might be nice. So, imagine an implementation that looks like:

```
const { imageCount, originalTotalSize, compressedImagesOriginalSize, compressedTotalSize, totalCompressedCount) = useStoreSelector(selectors.selectStats);
```

What do you think?

Reply:
Not sure I totally follow the example or benefit to doing it that way. Would you mind elaborating a bit?

Reply:
Simplifying consumption of data is one of the big benefits of reselect but often overlooked. That said, not everyone subscribes to this approach. But rather than having 5 different `select.getSomething` calls above, the idea would be to define 1 more selector that takes each of the above 5 and combines them into one return object that can easily be destructured up above. Not only is this highly reusable, but it's a really cool pattern in that it allow whatever data needs to be picked (whether it be 1, 2, or all 5 of the returned/nested selectors). I'm personally a fan of this approach as it cleans up the renderer, makes consumption easier, makes future refactors easier, and makes me feel good inside. It's not for everyone though, but that's the gist.

Reply:
Oh nice, that's rad!